plugins/inputs/vsphere/endpoint.go

```diff
@@ -679,7 +701,12 @@ func (e *Endpoint) Collect(ctx context.Context, acc telegraf.Accumulator) error
 			wg.Add(1)
 			go func(k string) {
 				defer wg.Done()
-				err := e.collectResource(ctx, k, acc)
+				var err error
+				if k == "vsan" {
+					err = e.collectVsan(ctx, k, acc)
```

Comment:
The issue is vSAN has its own APIs and endpoints, it can't fit into collectResource func. We've also thought of creating a separate vSAN plugin, but we will need to copy code for connect, discovery, finder, etc.